sklearn/linear_model/_omp.py

```diff
@@ -657,10 +658,11 @@ class OrthogonalMatchingPursuit(MultiOutputMixin, RegressorMixin, LinearModel):
     n_iter_ : int or array-like
         Number of active features across every target.
 
-    n_nonzero_coefs_ : int
+    n_nonzero_coefs_ : int or None
         The number of non-zero coefficients in the solution. If
         `n_nonzero_coefs` is None and `tol` is None this value is either set
         to 10% of `n_features` or 1, whichever is greater.
+        When `None`, `n_nonzero_coefs` has been ignored because `tol` has been set.
```

Comment:
And then maybe explain the internal logic of calculating `n_nonzero_coefs_` here:

```suggestion
        The number of non-zero coefficients in the solution. `None`, if `tol` is set, 
        otherwise is `n_nonzero_coefs` if this param is set. If both 
        `n_nonzero_coefs` and `tol` are `None`, this value is set to either 10% 
        of `n_features` or 1, whichever is greater.
```
I tried to rephrase so it's more intuitive. You can also ignore that, it it doesn't feel more intuitive to you.

Reply:
I see your point, I've re-word.